shared/dataflow/codeql/dataflow/TaintTracking.qll

```diff
@@ -71,7 +71,8 @@ module TaintFlowMake<
         Config::isSink(node) or
         Config::isSink(node, _) or
         Config::isAdditionalFlowStep(node, _, _) or
-        Config::isAdditionalFlowStep(node, _, _, _, _)
+        Config::isAdditionalFlowStep(node, _, _, _, _) or
+        defaultAdditionalTaintStep(node, _, _)
```

Comment:
^ for the benefit of reviewers, this is the actual change, everything else is consequences.

Reply:
It was left out on purpose to avoid lazy/imprecise models. Models from e.g. MaD are expected to handle content precisely without the need for implicit reads. I'm afraid that adding this could yield unintended FP flow in all sorts of places, but I don't know for sure. It's certainly risky, and I wouldn't expect it to be necessary.

Could you elaborate on the motivating example?
```
let _ = string1 + &string2;
```
Where's the missing read step - is it from `string2` to `&string2` or something like that? (please excuse my ignorance of Rust semantics). If so, it would seem that there would be plenty of room to add a proper read step.

Reply:
No, we have flow from `string2` to `&string2` and it adds `ReferenceContent`.  The problem is that our model for `+` (in `defaultAdditionalTaintStep`) expects there to be no content.  The compiler I believe adds an implicit dereference but there's nothing in our AST representing that.

Thus, I think we want to be able to read out of a `ReferenceContent` just about anywhere.  *or* we need to figure out where the implicit dereferences occur – but I suspect we don't have enough information about types in the AST for that yet.

Reply:
> we need to figure out where the implicit dereferences occur – but I suspect we don't have enough information about types in the AST for that yet.

If the key word in that sentence is "yet", then this sounds like the right approach.